src/value/mod.rs

```diff
@@ -167,6 +187,56 @@ pub(crate) fn mg_value_string(mg_value: *const bindings::mg_value) -> String {
     mg_string_to_string(c_str)
 }
 
+fn days_as_seconds(days: i64) -> i64 {
+    hours_as_seconds(days * 24)
+}
+
+fn hours_as_seconds(hours: i64) -> i64 {
+    hours * 60 * 60
+}
+
+fn minutes_as_seconds(minutes: i64) -> i64 {
+    minutes * 60
+}
+
+const NSEC_IN_SEC: i64 = 1_000_000_000;
+
+pub(crate) fn mg_value_naive_date(mg_value: *const bindings::mg_value) -> Result<NaiveDate, ()> {
+    let c_date = unsafe { bindings::mg_value_date(mg_value) };
+    let c_delta_days = unsafe { bindings::mg_date_days(c_date) };
+    let epoch_date = NaiveDate::from_ymd(1970, 1, 1);
+    let delta_days = Duration::days(c_delta_days);
+    Ok(epoch_date.checked_add_signed(delta_days).unwrap())
+}
+
+pub(crate) fn mg_value_naive_local_time(
+    mg_value: *const bindings::mg_value,
+) -> Result<NaiveTime, TryFromIntError> {
+    let c_local_time = unsafe { bindings::mg_value_local_time(mg_value) };
+    let c_nanoseconds = unsafe { bindings::mg_local_time_nanoseconds(c_local_time) };
+    let seconds = u32::try_from(c_nanoseconds / NSEC_IN_SEC)?;
+    let nanoseconds = u32::try_from(c_nanoseconds % NSEC_IN_SEC)?;
+    Ok(NaiveTime::from_num_seconds_from_midnight_opt(seconds, nanoseconds).unwrap())
```

Comment:
```suggestion
    Ok(NaiveTime::from_num_seconds_from_midnight(seconds, nanoseconds))
```

Same behavior (panic in case of invalid values) but this is cleaner.